src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -387,19 +399,271 @@ export class PatientViewPageStore {
 
     readonly myCancerGenomeData: IMyCancerGenomeData = getMyCancerGenomeData();
 
-    readonly mutationalSignatureData = remoteData({
-        invoke: async () => fetchMutationalSignatureData(),
+    // get mutational signature molecular profile Ids (contribution and confidence)
+    readonly mutationalSignatureMolecularProfiles = remoteData<
+        MolecularProfile[]
+    >(
+        {
+            await: () => [this.molecularProfilesInStudy],
+            invoke: () => {
+                return Promise.resolve(
+                    this.molecularProfilesInStudy.result.filter(
+                        (profile: MolecularProfile) => {
+                            if (profile.genericAssayType) {
+                                return (
+                                    profile.genericAssayType ===
+                                    GenericAssayTypeConstants.MUTATIONAL_SIGNATURE
+                                );
+                            }
+                            return false;
+                        }
+                    )
+                );
+            },
+        },
+        []
+    );
+
+    readonly fetchAllMutationalSignatureData = remoteData(
+        {
+            await: () => [
+                this.samples,
+                this.mutationalSignatureMolecularProfiles,
+            ],
+            invoke: () => {
+                let sampleMolecularIdentifiers: SampleMolecularIdentifier[] = [];
+                const mutationalSignatureMolecularProfileIds = this.mutationalSignatureMolecularProfiles.result.map(
+                    profile => profile.molecularProfileId
+                );
+                if (mutationalSignatureMolecularProfileIds.length > 0) {
+                    _.forEach(
```

Comment:
this is really a mapping operation i believe.  so sampleMolecularIndentiefiers = whatever.map()